### PR TITLE
update to cert-manager >= v1.0

### PR DESCRIPTION
**Pull request by MichaelSp**
WARNING: This is a breaking change. the `dns01provider` will no longer be provided as part of the `Certificate` but can be configured on the `Issuer`/`ClusterIssuer` level. This makes configuration of `Certificates` much easier.

Example `Issuer` configuration for `cloudDNS`:

```yaml
apiVersion: cert-manager.io/v1
kind: Issuer
metadata:
  name: letsencrypt-staging
spec:
  acme:
    server: https://acme-staging-v02.api.letsencrypt.org/directory
    email: <your mail>
    privateKeySecretRef:
      name: letsencrypt-staging
    solvers:
    - dns01:
        cloudDNS:
          project: my-project
          serviceAccountSecretRef:
            name: prod-clouddns-svc-acct-secret
            key: service-account.json
```

### CODE REVIEW

File: charts/docker-mailserver/README.md [hunk 1]
```diff
@@ -11,28 +11,27 @@ Kubernetes](https://github.com/docker-mailserver/docker-mailserver/wiki/Using-in
...
+  - [Setup OpenDKIM](#setup-opendkim)
+  - [Setup RainLoop](#setup-rainloop)
+  - [Configuration](#docker-mailserver-configuration)
+    - [Minimal configuration](#minimal-configuration)
+    - [Chart Configuration](#chart-configuration)
+    - [docker-mailserver Configuration](#docker-mailserver-configuration)
+    - [Rainloop Configuration](#rainloop-configuration)
+    - [HA Proxy-Ingress Configuration](#ha-proxy-ingress-configuration)
+- [Development](#development)
+  - [Testing](#testing)
```
Comment (MichaelSp) [inline charts/docker-mailserver/README.md:34]: Sorry, this change is kinda unrelated. I thought it makes sense the fix the links while I'm at it....

**Comment by funkypenguin** [comment]
The breaking change makes sense, it's ridiculous that our chart is so far behind in terms of cert-manager versions! However, if we're making a breaking change, convention dictates that we bump the semver major version in `Charm.yaml` accordingly. Would you please bump this, and resolve the current conflict?

Thanks for keeping us up-to-date! :)
D

**Comment by alexandre-hallaine** [comment]
really need this fix

**Review by funkypenguin — APPROVED** [review]
Apologies, missed this. Thanks for the major bump, looks good!

**Pull request merged**